kernel/src/drivers/ps2/mod.rs

```diff
@@ -3,82 +3,85 @@ pub mod io;
 use drivers::ps2::io::*;
 use spin::Mutex;
 
-pub const DEVICE_ENABLED_FLAG: u8 = 1 << 0;
-pub const DEVICE_SECOND_FLAG: u8 = 1 << 1;
-pub const DEVICE_AVAILABLE_FLAG: u8 = 1 << 2;
-
 pub const RESEND: u8 = 0xFE;
 pub const ACK: u8 = 0xFA;
-pub const SELF_TEST_PASSED: u8 = 0xAA;
 
-pub static PS2: Mutex<Ps2Controller> = Mutex::new(Ps2Controller::new());
+lazy_static! {
+    pub static ref PS2: Mutex<Ps2Controller> = Mutex::new(Ps2Controller::new());
+}
+
+bitflags! {
+    pub struct Ps2ConfigFlags: u8 {
+        const PORT_INTERRUPT_1 = 1 << 0;
+        const PORT_INTERRUPT_2 = 1 << 1;
+        const SYSTEM = 1 << 2;
+        const KEYBOARD_LOCK = 1 << 4;
+        const PORT_OUTPUT_FULL_2 = 1 << 5;
+        const PORT_TRANSLATION_1 = 1 << 6;
+    }
+}
 
 /// Represents the PS2 master controller
 pub struct Ps2Controller {
     pub initialized: bool,
```

Comment:
Do we really need this? (Not really a change but could be changed)

Reply:
For safety, you don't want the controller to be initialized twice.

Reply:
Ah OK. Then shouldn't we test it in the initialize function too?

Reply:
Yep, we should. I believe you addressed this in another comment.